components/DatabaseExplorer/TablesActionsList.tsx

```diff
@@ -28,45 +32,55 @@ export const TablesActionsList = ({
     action: activeAction,
   } = useDatabaseParams()
 
+  const [activeProcedure, setActiveProcedure] = useState<string | null>(null)
+
   return (
     <>
-      {items.map((objectItem: KwilTypes.Table | KwilTypes.ActionSchema | Procedure) => (
-        <div
-          data-testid={`database-item-${dbid}-${itemType}-${objectItem.name}`}
-          key={`${dbid}-${itemType}-${objectItem.name}`}
-          className="ml-6 overflow-hidden text-sm"
-        >
-          <Link
-            href={`/databases/${dbid}/${itemType === ItemTypes.PROCEDURES || itemType === ItemTypes.ACTIONS ? `method/` : ``}${itemType.slice(0, -1)}/${objectItem.name}`}
-            className={classNames({
-              "flex select-none flex-row items-center gap-1 hover:text-slate-900":
-                true,
-              "text-slate-500 ": !isTableOrActionActive(
-                dbidParam,
-                dbid,
-                itemType,
-                activeTable,
-                activeAction,
-                objectItem.name,
-              ),
-              "font-semibold text-slate-900": isTableOrActionActive(
-                dbidParam,
-                dbid,
-                itemType,
-                activeTable,
-                activeAction,
-                objectItem.name,
-              ),
-            })}
-            onClick={() => {
-              dispatch(setIsMenuOpen(false))
-            }}
+      {items.map(
+        (objectItem: KwilTypes.Table | KwilTypes.ActionSchema | Procedure) => (
+          <div
+            data-testid={`database-item-${dbid}-${itemType}-${objectItem.name}`}
+            key={`${dbid}-${itemType}-${objectItem.name}`}
+            className="ml-6 overflow-hidden text-sm"
           >
-            <ChevronRightIcon className="h-3 w-3" />
-            <span className="max-w-[80%]">{objectItem.name}</span>
-          </Link>
-        </div>
-      ))}
+            <Link
+              href={`/databases/${dbid}/${itemType === ItemTypes.PROCEDURES || itemType === ItemTypes.ACTIONS ? `method/` : ``}${itemType.slice(0, -1)}/${objectItem.name}`}
+              className={classNames({
+                "flex select-none flex-row items-center gap-1 hover:text-slate-900":
+                  true,
+                "text-slate-500": !isTableOrActionActive(
+                  dbidParam,
+                  dbid,
+                  itemType,
+                  activeTable,
+                  activeAction,
+                  objectItem.name,
+                ),
+                "font-semi-bold text-slate-900": isTableOrActionActive(
+                  dbidParam,
+                  dbid,
+                  itemType,
+                  activeTable,
+                  activeAction,
+                  objectItem.name,
+                ),
+                "font-bold text-slate-900": itemType === ItemTypes.PROCEDURES && activeProcedure === objectItem.name
```

Comment:
@tydangelo18 - there are two issues with the approach you have taken:

1. It only highlights procedures. This still leaves actions unhighlighted.
2. When the user navigates away from procedures (e.g., to a table or action), the previous procedure they selected remains highlighted.

The correct approach would have been to adjust `isTableOrActionActive` to account for procedures and actions. I will push up changes for this in a minute.